networks/hamiltonian_net.py

```diff
@@ -0,0 +1,56 @@
+"""This module contains an implementation of the Hamiltonian network described in the paper.
+The Hamiltonian network takes the abstract positions and momenta, q and p, and computes a scalar
+value that is interpreted as the Hamiltonian.
+"""
+
+import torch
+from torch import nn
+
+
+class HamiltonianNet(nn.Module):
+    """The Hamiltonian network, composed of 6 convolutional layers and a final linear layer.
+    """
+
+    def __init__(self, in_channels, dtype=torch.float):
+        """Create the layers of the Hamiltonian network.
+
+        Args:
+            in_channels (int): Number of input in_channels.
+            dtype (torch.dtype): Type of the weights.
+        """
+        super().__init__()
+        self.num_flat_features = 64 * 4 * 4
+        self.in_conv = nn.Conv2d(in_channels=in_channels, out_channels=64, kernel_size=3, padding=1)
```

Comment:
Again, I would like the hardcoded parameters and number of layers to be arguments.

I would like the architecture to be easy to define from a yaml file so that testing different architectures becomes very easy. Maybe we dont need it for now but I think it would be very good to have for when we need to experiment.

Reply:
Done as we discussed